### DISCUSSION TITLE
Database migrations workflow for live iOS app?

### DISCUSSION THREAD

**Discussion by garygcchiu** [Nozbe/WatermelonDB#1770] [Q&A]
I have an app using React Native, and WatermelonDB. 

What is the recommended way to adding a new db migration? The [docs](https://watermelondb.dev/docs/Advanced/Migrations) mention refreshing my simulator/browser, but that doesn't make sense for live users since we are required to submit the binary to Apple -> await their review -> users install. 

Is this the recommended workflow?

1. Add new database migration
2. Release (submit to App Store -> go through Apple review process)
3. Make matching change in schema
4. Release
5. Bump schema version
6. Release

The problem I see with this approach is there is no guarantee all users will incrementally use the latest version. If I have app:
- Shipped version `v1.1` with database version `1`
- Then ship `v1.2` with database version `2` w/ a new migration added
- Then ship `v1.3` with matching schema for database version `2` added
- Then bump schema version to `2`

In the above scenario, if the user is on v1.1 and then downloads v1.3, wouldn't their database reset? 

Is there a recommended approach for deploying out a database migration?

**Comment by KrisLau** [comment]
> In the above scenario, if the user is on v1.1 and then downloads v1.3, wouldn't their database reset?

It wouldn't because the migrations on the migration should have incremental steps from 1.1 to 1.2 to 1.3. For example, version 1.1 would have:
```
export default schemaMigrations({
  migrations: [
    {
      toVersion: 2,
      ...
    },
  ],
})
```
and version 1.3 would have:
```
export default schemaMigrations({
  migrations: [
    {
      toVersion: 3,
      ...
    },
    {
      toVersion: 2,
      ...
    },
  ],
})
```
Refer to this: [https://watermelondb.dev/docs/Advanced/Migrations#migrations-api](https://watermelondb.dev/docs/Advanced/Migrations#migrations-api)

That being said I wish there was something like an `onSchemaVersionChanged` where I could just reset the whole DB and sync from the backend again. If anyone knows anything about that I would appreciate any help here: https://github.com/Nozbe/WatermelonDB/issues/1609
